### ISSUE TITLE
wrong missing whitespace warning

### ISSUE THREAD

**Issue by matthieu-ravey** [palantir/tslint#463]
I have a missing whitespace warning on lines like these:
`$scope.datasource.datasourceVersion = $scope.datasource.datasourceVersion || 2;`

**Comment by adidahiya** [comment]
are you on the latest tslint?

**Comment by matthieu-ravey** [comment]
I'm using grunt-tslint@2.0.0 and tslint@2.1.1

**Comment by adidahiya** [comment]
please try with the latest versions; both grunt-tslint and tslint are on version `2.3.0-beta` now and have bugfixes that might solve your issue.

**Comment by matthieu-ravey** [comment]
I updated grunt-tslint and tslint to 2.3.0-beta and I still have the issue.

**Comment by leeavital** [comment]
1. Does running standalone tslint (without grunt) still cause the problem?

2. What rules are you using? Can you post your tslint.json?

**Comment by AlaricCalmette** [comment]
Hi! I work with Matthieu.
We are using the rule Whitespace like this : 
"whitespace": [
			true,
			"check-branch",
			"check-decl",
			"check-operator",
			"check-separator",
			"check-type"
		]

I've looked through the WhiteSpace walker code and I logged the type of the token responsible for the failure.

From what I saw, the problem comes from the fact that after a 'barbartoken'  the function CheckForTrailingWhitespace is called with a position one character too far to the right.

I explain : 
If i have : 
temp1 || 8 
The type of the token creating the failure is numericLitteral  

If i have :
temp2 || temp1
The type of the token is Identifier

So it's either the position that is false or the token barbar is "|| " instead of "||"

I'll keep looking and keep you posted.

**Comment by AlaricCalmette** [comment]
Any update on this?

**Comment by JKillian** [comment]
I'll look into it today or tomorrow and get back to you

**Comment by AlaricCalmette** [comment]
Okay thanks!

**Comment by JKillian** [comment]
I can't seem to reproduce this. I ran TSLint on a file that looked like this:

```typescript
let $scope: any;
$scope.datasource.datasourceVersion = $scope.datasource.datasourceVersion || 2;
console.log($scope);
```
with the whitspace rule set as such:
```
    "whitespace": [true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ]
```

Do you get the whitespace error on this code?

**Comment by AlaricCalmette** [comment]
Hum... My bad. In fact those false whitespace were unbreakable space. So neither " " nor "\s". Replacing those characters with 'true' space fix the issue.

**Comment by JKillian** [comment]
Sweet, glad it got figured out!
